GS-61 add total quantity of item stored in db

VaolEr opened the pull request:
Добавил это в функцию get, по причинам:
1. От фронтовиков поступила просьба, чтоб в общем списке сразу летел суммарный остаток по всем складам, а детальная информация по /items/id.
2. Для того чтоб не плодить методы и RequestParams.

Возможно там как-то более изящно это можно подсунуть, но я сделал так и оно работает. Где-то возможно не доглядел что-то.

VaolEr committed:
Add totalQty field to ItemsService response;

VaolEr committed:
Add totalQty field to ItemsService response;

VaolEr committed:
Merge branch 'dev' into GS-61-add-total-quantity-of-item-stored-in-db

dimio commented on src/main/java/com/example/storehouse/dto/ItemTo.java:
```diff
@@ -50,4 +50,7 @@
         example = "[{\"id\": 1000, \"quantity\": 25}, {\"id\": 1001, \"quantity\": 15}]")
     List<ItemStorehouseTo> itemsStorehouses;
 
+    @NotNull
```
Эта же ТОшка принимается для создания item, и там такого поля не будет. Проверял запрос на добавление item?

VaolEr replied:
Не проверял(

VaolEr replied:
Пофиксил все проблемы, связанные с добавлением общего числа товаров=)

dimio commented on src/main/java/com/example/storehouse/repository/StorehousesRepository.java:
```diff
@@ -21,7 +22,7 @@
-    @Query("select i_s.quantity from ItemStorehouse i_s where i_s.item.id = :itemId")
-    Integer getQuantityByItemId(Integer itemId);
+    @Query(value = "SELECT SUM(i_s.quantity) FROM ItemStorehouse i_s WHERE i_s.item.id = :itemId")
+    Integer getQuantityByItemId(@Param("itemId") Integer itemId);
```
@Param можно не указывать в данном случае.

VaolEr committed:
Add ValidationGroups and UnitsRepository;
Changed validation method for create method in ItemsController;
Changed prepareToSave method in ItemsService;

VaolEr committed:
Fixed ItemsServiceTest;

VaolEr committed:
Updated README.md and pom.xml (SNAPSHOT);

dimio commented on src/main/java/com/example/storehouse/dto/UnitTo.java:
```diff
@@ -18,8 +18,8 @@
-    @NotNull
-    @NotBlank
+    //@NotNull
+    //@NotBlank
```
А что тут за проблема была?
